packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2046,33 +2051,36 @@ export class SnapController extends BaseController<
    */
   async getSnapState(snapId: SnapId, encrypted: boolean): Promise<Json> {
     const runtime = this.#getRuntimeExpect(snapId);
-    const cachedState = encrypted ? runtime.state : runtime.unencryptedState;
+    return await runtime.getStateMutex.runExclusive(async () => {
```

Comment:
With this cached requests would still have to queue to access the cached, right? That seems less than ideal 🤔

Reply:
Would it work if we have a `cacheMutex` that nests this one?

Reply:
What do you mean by nesting in this case?

> With this cached requests would still have to queue to access the cache, right? That seems less than ideal 🤔

Yeah, but cached access should be much quicker, so I don't think it's a big issue?

Reply:
Let's try this as-is and we can always move some logic outside of the mutex block